iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorNonLinearModelImpl.java

```diff
@@ -49,19 +49,46 @@ public double getG() {
 
     @Override
     public double getMaximumB() {
-        return sections.values().stream()
-                .map(SectionImpl::getB)
-                .max(Double::compare)
-                .orElseThrow(() -> new PowsyblException("a shunt compensator must have at least one section"));
+        if (sections.values().stream().allMatch(section -> section.getB() >= 0)) {
+            return sections.values().stream().mapToDouble(SectionImpl::getB).sum();
+        } else if (sections.values().stream().allMatch(section -> section.getB() <= 0)) {
+            return 0;
+        }
+        return sections.keySet().stream().mapToDouble(this::getCurrentB).max().orElseThrow(() -> new PowsyblException("a shunt compensator must have at least one section"));
```

Comment:
I think it's not correct:
- if all values are positive, return the sum of all positive values
- if all values are negative, return 0
- if values are positve either negatives, you have to sum all positive values (general case).

From my point of view, the general case works for all cases...

Reply:
Modified as the maximum between 0 and the sum of all susceptances.